### ISSUE TITLE
Have a "learn more" button beside each feature on Optimole that leads to the relevant documentation page

### ISSUE THREAD

**Issue by Nazrinn** [Codeinwp/optimole-wp#621]
### What problem does this address?

Sometimes the option description isn't enough to know what it does in details. Also I didn't know Optimole had actually good documentation about each option until today.

### What is your proposed solution?

At the end of an option's description, add a "learn more" link that leads to the option's detailed documentation, if it exists. Similar to what the LiteSpeed cache plugin does (as shown below).

![image](https://github.com/Codeinwp/optimole-wp/assets/36900693/b9dd8784-241a-42d5-8248-1ebc64af2a5b)

### Will this feature require documentation? (Optional)

No.

___
### [Figma](https://www.figma.com/file/om1c4NvIs6ahzkZVSUw4a6/Optimole-WP-Plugin?type=design&node-id=4719-92267&mode=design&t=lFIlabs32xK89xwg-11)

**Comment by selul** [comment]
Good point, @Nazrinn! Definitely valuable to have for settings that are more complex. 

Out of curiosity, could you please elaborate on which specific setting or option you believe would benefit from this? 

Thank you!

**Comment by Nazrinn** [comment]
Of course!

Here are some settings I think could use a link for some extra information from your documentation and why:

**General**
- Scale images & Lazy load
_Perhaps a link that details how image size is measured? This is mostly fine as is though._
- Enable error diagnosis tool
_What exactly does this setting do to the site?_
- Clear cached resources 
_What exactly does this setting do to the site and/or images?_

Bonus: For "Enable image replacement", it would be great if you specified you serve images through your CDN, in the description.


**Advanced > Compression**
- Enable network based optimizations 
_How heavily does it impacts quality? Is there a minimum quality?_
- Strip Image Metadata
_Detail how this reduces page load._


**Advanced > Compression**
- Enable Auto Quality powered by ML(Machine Learning)
_What does that do and when would turning this off be ideal?_


**Advanced > Resize**
- Enable Smart cropping
_Detail how this work and in what circumstances this applies/does not apply._
- Enable Retina images
_This setting should specify what retina images are. I'm sure most people don't know._
- Add a new crop image size
_I don't really understand the description. I'm not sure what this does._
- Resize large images original source.
_The description is not very clear. I'm not sure what this does and how it differs from "Limit Image Dimensions with max width/height" or "Add a new crop image size"._


**Advanced > Lazyload**
- Enable generic lazyload placeholder
_What is a generic lazyload placeholder? What does that do? How does it affect the layout shift core web metric? (This is the setting that made me start this thread.)
- Scale Images
_How does Optimole scale images and which settings are connected to this one?_
- Lazyload background images for selectors:
_Do I need to fill that up if "Enable lazyload for background images" is checked? Are the selectors listed exceptions to "Enable lazyload for background images"?

**Cloud Integration**
- Enable offloading images
_I'd like to see details on how this affect the site. Positively or negatively. I'm not sure why the yellow box only appear when that option is enabled._

Reminder that I think most short option descriptions are fine as-is, unless I specified something was unclear. This is information that at least should be stated in the linked documentation.


**Bonus**
Since I've gone through the settings one -by-one, I spotted some mistakes.

These settings are missing a period at the end of their description:
- Enable Gif to Video conversion
- Enable avif conversion
- Enable Retina images
- Use native lazyload
- Enable lazyload for embedded videos and iframes.

These settings have a space between the last word and period of their description:
- Scale images & Lazy load

These settings have a period at the end of their name:
- Resize large images original source.
- Enable lazyload for embedded videos and iframes.

**Comment by selul** [comment]
Wow, thanks a lot for taking the time for writing those, they are really helpful. We will take care of them.

**Comment by Nazrinn** [comment]
My clients are out on vacation and I have a lot of time to lose. HAH!

**Comment by mghenciu** [comment]
> https://github.com/orgs/Codeinwp/teams/design-team do you see any opportunities to enhance the UI in this regard? Maybe we can have a more cohesive approach with the documentation links rather than a simple learn more link?

**Maybe we can use an extended tooltip**. Basically a tooltip is a smaller informative element, that doesn't have any interactive elements.

<img width="1797" alt="image" src="https://github.com/Codeinwp/optimole-wp/assets/52494172/83179311-92c7-4e4e-a154-0df8541abe48">


**While an Extended Tooltip or Toggletip, is a bigger element that is opened on click/tap.**
Inspired from [Carbon Design System](https://carbondesignsystem.com/components/toggletip/usage), also [wikipedia](https://en.wikipedia.org/wiki/Aristotle) tooltips.

The Toggletip should follow the [best practices from NN Group](https://www.nngroup.com/articles/tooltip-guidelines/), meaning that if the information **is necessary for a user to complete the task** -> it should be visible direcly in the main view, otherwise we can use a Toggletip/Tooltip.

**Comment by Nazrinn** [comment]
I think the link is absolutely important given that a documentation description can be as detailed as needed. Without it as a user, I also would definitely not know Optimole has detailed documentation.

Personally I'm not sure about the "tooltip" thing given that the space in a tooltip would be much more limited compared to the current description and could be awkward to implement on smaller screens.

As a user, I almost always read the description and I feel that it's nice to have there rather than having to click/hover something to see it.

**Comment by selul** [comment]
@AndreeaCristinaRadacina and others I think for this, we will need to re-organize the docs and have individual Documentation pages for each setting detailing thourghly each one of the option and focusing on: 

- A more detailed description
- The impact that will have to the user site/images/etc
- The benefits or trade-offs that the user needs to make.
- More helpful links from external sources. I.e for Retina like https://en.wikipedia.org/wiki/Retina_display


@JohnPixle I'm also not that convinced by the tooltip there vs the regular description like we have now for the reason that: 
- users need to do an extra click
- it might not be obvious that you can find additional info if you click on that link. 
- Accessibility is becoming more complex when using tooltips vs regular description texts.

**Comment by AndreeaCristinaRadacina** [comment]
Hi @selul,

I have already started re-organizing the docs. What do you think about having separate docs for each setting panel?
You can check a preview here - https://docs.optimole.com/article/1927-resize-optimole-advanced-settings?preview=650b048232eb6f503224dfce

In this way, all settings corresponding to a panel will be in the same place, and each setting will have a different anchor - https://docs.optimole.com/article/1927-resize-optimole-advanced-settings?preview=650b048232eb6f503224dfce#retina-images

Otherwise, I can create separate docs, as you suggested.
Let me know what you think!

**Comment by selul** [comment]
@AndreeaCristinaRadacina Yes, I think each setting should have its doc if we plan to offer a more detailed explanation for each one and link them near each option inside the plugin.

**Comment by selul** [comment]
I would recommend also to share 1-2 docs first and agree on the structure and after we can go on with all. In case that there are corrections needed, we can avoid correcting all of them vs just 1/2.

**Comment by mghenciu** [comment]
hey, I am back with an update here. After a nice teamwork between me and @AndreeaCristinaRadacina , we have a list with suggestions regarding text. Would love to hear you feedback, if any. We'll try to have a Learn more for most of the features (Andreea is working on the docs now).

And also it would help to have some clarifications from the @Codeinwp/optimole team regarding a few things (added below):
- how/what the Scale Images features does, from the Lazyload section?
- for the Image Replacement we want to use this: _Replace all the image URLs from your website with the ones optimized by Optimole, which also serves them from CDNs across the globe, ensuring lightning-fast loading times._  And wanted to confirm, is the CDN part true, right?
- for the Enable offloading images we use this Alert: _You are not required to use the offload functionality for the plugin to work, use it if you want to save on hosting space._ Is this message required? why do we show it?


### [New Descriptions in Notion](https://www.notion.so/themeisle/Optimole-descriptions-8467e37579ed45738905112323ed0f77?pvs=4)

**Comment by selul** [comment]
> how/what the Scale Images features does, from the Lazyload section?

Usually, when you open a webpage, all the images load at once, taking up time and data. With "lazyload," the images appear only when you scroll down to where they are placed on the page.

Optimole's feature normally does two things: it "lazyloads" images and resizes them to fit your screen better. However, this specific setting allows you to turn off that automatic resizing while keeping the lazyload function active. In simple terms, images will still appear only when you scroll down to them, but they will remain in their original size, not resized to fit your device's screen

This means that if an image in your content is originally 2000x3000 pixels but displays on a browser at a much smaller size like 200x300 pixels, disabling the resize would load the image at its full 2000x3000 pixels when it appears on the screen.

Impact of Disabling Automatic Scaling:

- True-to-Source Display: When you disable automatic resizing, your images will display in their full original dimensions, showing the content as it was created.
- Maximum Image Quality: The images will appear in their highest quality since they are not being compressed or resized.
- Design Consistency: If you have specific images like infographics, where fine details matter, disabling the resize feature ensures these images are displayed at their best. 

Tradeoffs:

- Oversized Images: With the scaling turned off, that 2000x3000 pixel image will appear much larger than the 200x300 pixel display box. This could lead to a messy layout and require users to scroll to see the entire image.
- Increased Data and Bandwidth Use: When images load at their full size, they consume more data, which might be an issue for users on limited data plans.
- Slower Page Load: Larger images take longer to load, especially on slower connections or older devices, which could impact user experience.
 
> for the Image Replacement we want to use this

Image replacement is the option basically that turn on/off Optimole. Maybe a better name would be: `Enable Optimole Image Handling`

The "Image Replacement" button in Optimole is essentially your master switch for Optimole's image services. When activated by toggling it "On," this feature hands over the responsibility of serving all the images across your website to Optimole's specialized service. This includes features like lazyloading, resizing, and even compressing your images to make them load faster.
Turning the feature "Off" will revert the image management back to its original state. This means your website will serve images from their original locations.
So, in a nutshell, the "Image Replacement" feature is your master control for letting Optimole manage, optimize, and serve all the images on your website. 


> for the Enable offloading images, we use this Alert: You are not required to use the offload functionality for the plugin to work, use it if you want to save on hosting space. Is this message required? why do we show it?

I don't think so if the description is clear on what it does.

**Comment by mghenciu** [comment]
Thank you for very detailed explanation, @selul , it really helped.

Regarding the Image Replacement feature, as you mentioned that this is the master switch for Optimole image services - does this also means that other options shouldn't be available/wont work, in case the toggle is turned off?

for example let's say **I have the Enable image replacement turned OFF** - then the Compression, Resize, Lazyload panels will still produce any changes or not to my site. If not, I am thinking we can show a notice/disabled state for those features.

**Comment by selul** [comment]
@mghenciu Yes, in fact, this was the behavior but it might have gotten lost during the react migration, please check version 3.7.0 to see this, available here https://wordpress.org/plugins/optimole-wp/advanced/

**Comment by AndreeaCristinaRadacina** [comment]
Hi there,

I prepared a doc following the suggested structure:
https://docs.optimole.com/article/1871-how-to-use-the-smart-cropping-in-optimole?preview=64a558853f11c3087d399770

Also, a template is available in our internal workarounds, which I plan to use for all features if you agree with it:
https://docs.themeisle.com/article/1931-optimole-features (it looks slightly different on the Optimole docs website, as you can see in the above doc, but focus on the structure).

Let me know what you think!

**Comment by selul** [comment]
Hi @AndreeaCristinaRadacina,

- I would suggest keeping the same title structure for all, on one we use the `How to use the ...` on the other, we use 'Exclude from ..', we can keep the same one, i.e `Smart Cropping` 
- The note does not make sense https://vertis.d.pr/i/gDmeWH and it complicates the doc. 
- This is related to another setting is has nothing to do with Smart Crop: https://vertis.d.pr/i/1gRhUs
- This has nothing to do with the doc https://vertis.d.pr/i/4M8pY1, it should be a separate one with custom codes, i.e `How to force cropping to images` or something.
-  The benefits are a bit too vague, i.e https://vertis.d.pr/i/E5GqKo, I don't see how Smart Cropping drives the Responsiveness benefit. Same for others.

Also, I want to point out that Benefits/Tradeoffs is more like a suggestion, we don't need to have those sections if we don't have anything to put there. 

We can also have a section called Use Cases, i.e when this setting is particularly useful to have. 

We can also have a new section, if we have data to add for it, Related Settings, where we can include links to the other settings that are relevant here, like we had with Custom crop size. Also, we can inline link them in the text aswell.

**Comment by AndreeaCristinaRadacina** [comment]
Hi @selul,

Thank you for your valuable feedback!
I've made the suggested adjustments to the smart cropping doc, which you can check [here](https://docs.optimole.com/article/1871-how-to-use-the-smart-cropping-in-optimole?preview=64a558853f11c3087d399770)

And I've updated the template as well, which you can find [here](https://docs.themeisle.com/article/1931-optimole-features)

Regarding the titles, I'm thinking we can keep it simple and stick to the feature names. For example, the "Enable Error Diagnosis tool" feature could be titled "Error Diagnosis Tool." 

What do you think? 😊

**Comment by selul** [comment]
@AndreeaCristinaRadacina for the titles is a good idea. 
Regarding the structure, my idea for related settings was to provide an external links to another documents with explanations on that, not embed that setting there. 

Also, I would recommend reviewing the Smart Cropping description since is not too intuitive. The one I've shared on Slack would be better: 

>The "Smart Cropping" feature in Optimole automatically identifies the most important or "interesting" part of an image. When the image is resized or needs to fit into a different shape or dimension, this feature ensures that the interesting part stays in the center and is visible, rather than getting cut off.

**Comment by mghenciu** [comment]
For the Descriptions in general, we were discussing with Andreea that it would help to have a consitenty like: 

- Feature Title (1st row)
- Description (2nd row below) 
- buttons/other panels (3rd row)

since I think the normal flow is to read the title + description (understanding the feature) - and after that clicking.


Currently here are some examples where we have the descriptions not under the Title.
<details><summary>Images</summary>

![image (5)](https://github.com/Codeinwp/optimole-wp/assets/52494172/09c30bf5-1176-408b-ac1b-980216c5d346)
![image (4)](https://github.com/Codeinwp/optimole-wp/assets/52494172/ead8ec30-9db1-4ab0-aa85-6abeefe16cc6)
![image (2)](https://github.com/Codeinwp/optimole-wp/assets/52494172/69009b75-e8b1-4d8d-bf26-592b75276c12)
![image (1)](https://github.com/Codeinwp/optimole-wp/assets/52494172/7078e535-d78a-405a-b6f9-8220b282d2fa)
![image](https://github.com/Codeinwp/optimole-wp/assets/52494172/e5265334-82fb-4456-aee7-468c28e6064f)


</details>

**Comment by Nazrinn** [comment]
Agreed that the button being just under the title looks odd.

**Comment by abaicus** [comment]
@HardeepAsrani Here, let's change the settings titles and descriptions to the ones provided [here](https://www.notion.so/themeisle/Optimole-descriptions-8467e37579ed45738905112323ed0f77), and also add the **Learn more** links. 

Let's also address what @mghenciu noted in [this comment](https://github.com/Codeinwp/optimole-wp/issues/621#issuecomment-1775077654), so we have a more consistent visual hierarchy across the settings.
